### ISSUE TITLE
Announce volume changes when using hardware buttons to adjust

### ISSUE THREAD

**Issue by levtecaccess** [nvaccess/nvda#5726]
Please enable NVDA to announce volume changes when using hardware buttons such as keyboard buttons or tablet PC volume rockers to raise or lower the system volume.

**Comment by jcsteh** [comment]
One problem is that this could get quite annoying if you're trying to 
adjust the volume while audio is playing. If NVDA was speaking, it would 
interrupt speech just to report the volume change. If audio is playing, 
it's enough to hear the change in volume rather than speaking a level.

**Comment by zahyur** [comment]
What about beeps? Although that could and I think should be done on the OS level. However, if there is a visual indication on the screen and no sound indication, that puts a screen reader user in an inferior position.
Perhaps one could have the choice between Speech, Beeps and None.

**Comment by levtecaccess** [comment]
Hi

I personally think that some kind of notification would help especially tablet users like myself.

**Comment by dkager** [comment]
Would a 'speak volume' script not be enough? This is what sighted people get by means of the volume icon in the system tray. Perhaps not as useful on a tablet, but I think it would suffice for PC users.
Seeing the actual volume changes in braille sounds potentially useful, but I wouldn't want a say-all to be interrupted for these speech announcements.

**Comment by bhavyashah** [comment]
I reckon a subtle ascending and descending tone of beeps would suffice for indication of volume changes by the volume rocker or other hardware buttons.

**Comment by dkager** [comment]
Actually, I would really like to hear the exact percentage spoken when such buttons are pressed. But I feel this borders on what a screen reader should support.

**Comment by levtecaccess** [comment]
I personally would like to have user configurable options to control what is heard when pressing volume buttons.

Best regards
Levi Johnson
Levtec Corporation
http://levtec.linkpc.net/

Sent from Mail for Windows 10

**Comment by leonardder** [comment]
From Windows 10 May 2019 update, this is actually implemented by Microsoft. Volume changes are reported when Windows Explorer has focus.

**Comment by leonardder** [comment]
I'm closing this as works for me as per above comment. If there is real desire for this feature on other versions of Windows, I'd like to suggest filing another issue for it with proper justification (i.e. why this belongs in NVDA core and not in an add-on).
